starlette/concurrency.py

```diff
@@ -50,8 +50,8 @@ def _next(iterator: typing.Iterator[T]) -> T:
 
 
 async def iterate_in_threadpool(
-    iterator: typing.Iterator[T],
-) -> typing.AsyncIterator[T]:
+    iterator: typing.Iterable[T],
+) -> typing.AsyncIterable[T]:
     while True:
         try:
             yield await anyio.to_thread.run_sync(_next, iterator)
```

Comment:
But `_next()` receives an Iterator... 👀

Reply:
I guess `mypy` didn't realize this, great catch! Looks like https://github.com/encode/starlette/pull/2357 updates to latest `mypy`, but unfortunately `scripts/check` still doesn't catch this

Reply:
Alright the fix was simple, having `iterate_in_threadpool` call `iter` before using `_next`. PR is ready for review again